snappy/kernel.go

```diff
@@ -0,0 +1,118 @@
+package snappy
+
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ubuntu-core/snappy/partition"
+	"github.com/ubuntu-core/snappy/pkg"
+	"github.com/ubuntu-core/snappy/pkg/squashfs"
+	"github.com/ubuntu-core/snappy/progress"
+)
+
+// noramlizeAssetName transforms like "vmlinuz-4.1.0" -> "vmlinuz"
+func normalizeKernelInitrdName(name string) string {
+	name = filepath.Base(name)
+	return strings.SplitN(name, "-", 2)[0]
+}
+
+func extractKernelAssets(s *SnapPart, inter progress.Meter, flags InstallFlags) (name string, err error) {
+	if s.m.Type != pkg.TypeKernel {
```

Comment:
Indeed, I did it out of laziness so that I would not have to check the SnapPart type before calling this function. That is a very poor reason of course so I changed it as you suggested.